dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java

```diff
@@ -56,41 +58,50 @@ public FileAssetAjax() {
-		WebContext ctx = WebContextFactory.get();
-		HttpServletRequest req = ctx.getHttpServletRequest();
-		User user = userAPI.getLoggedInUser(req);
-		boolean respectFrontendRoles = userAPI.isLoggedToFrontend(req);
-		Contentlet cont  = APILocator.getContentletAPI().find(contentletInode, user, respectFrontendRoles);
-		FileAsset fa = APILocator.getFileAssetAPI().fromContentlet(cont);
+	public void saveFileText(final String contentletInode, final String newText, final String binField)
+			throws PortalException, SystemException, DotDataException, DotSecurityException, IOException {
+
+		final WebContext webContext        = WebContextFactory.get();
+		final HttpServletRequest request   = webContext.getHttpServletRequest();
+		final User user 			       = userAPI.getLoggedInUser(request);
+		final boolean respectFrontendRoles = userAPI.isLoggedToFrontend(request);
+		final Contentlet contentlet        = APILocator.getContentletAPI().find(contentletInode, user, respectFrontendRoles);
+		String incomingFileName            = null;
+		String binaryFieldName             = binField;
+
+		if (contentlet.isFileAsset()) {
+
+			final FileAsset fileAsset = APILocator.getFileAssetAPI().fromContentlet(contentlet);
+			incomingFileName = fileAsset.getFileAsset().getName();
+			binaryFieldName  = FileAssetAPI.BINARY_FIELD;
+		} else if (contentlet.isDotAsset()) {
+
+			incomingFileName = contentlet.getBinary(DotAssetContentType.ASSET_FIELD_VAR).getName();
+			binaryFieldName  = DotAssetContentType.ASSET_FIELD_VAR;
+		} else {
+
+			incomingFileName = contentlet.getBinary(binField).getName();
+		}
 
-		java.io.File tempDir =  new java.io.File(APILocator.getFileAssetAPI().getRealAssetPathTmpBinary() + java.io.File.separator + contentletInode.charAt(0)
-					+ java.io.File.separator + contentletInode.charAt(1) + java.io.File.separator + contentletInode
-					+ java.io.File.separator + APILocator.getFileAssetAPI().BINARY_FIELD);
+		final File tempDir =  new File(APILocator.getFileAssetAPI().getRealAssetPathTmpBinary() + File.separator + contentletInode.charAt(0)
+				+ File.separator + contentletInode.charAt(1) + File.separator + contentletInode
+				+ File.separator + binaryFieldName);
 
-		if(!tempDir.exists())
+		if(!tempDir.exists()) {
 			tempDir.mkdirs();
+		}
 
-		java.io.File fileData = new java.io.File(tempDir.getAbsoluteFile() + java.io.File.separator + WebKeys.TEMP_FILE_PREFIX + fa.getFileAsset().getName());
+		final File fileData = new File(tempDir.getAbsoluteFile() + File.separator + WebKeys.TEMP_FILE_PREFIX + incomingFileName);
 
 		fileData.deleteOnExit();
-		OutputStream os = null;
-		try {
-			os = Files.newOutputStream(fileData.toPath());
+
+		try (OutputStream os = Files.newOutputStream(fileData.toPath())) {
+
 			os.write(newText.getBytes());
 		} catch(Exception e) {
-			Logger.error(getClass(), "Error writing to file", e);
-		}finally {
-			if (os != null){
-				os.close();
-			}
 
+			Logger.error(getClass(), "Error writing to file", e);

```

Comment:
should this logic get into a API?

Reply:
I would probably promote to an endpoint, I might create an issue

Reply:
https://github.com/dotCMS/core/issues/18421